### PR TITLE
[CDAP-12944] add macro widgets

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-12944

### CODE REVIEW

File: cdap-ui/app/hydrator/templates/partial/node-config-modal/node-config/plugin-edit-properties-template.html [hunk 1]
```diff
@@ -41,107 +41,29 @@ <h4>{{::group.display}}</h4>
...
-              </span>
-            </div>
           </div>
-        <div class="form-group">
-          <div class="col-xs-offset-8 col-xs-4">
+
+          <macro-widget-toggle
+            node="HydratorPlusPlusNodeConfigCtrl.state.node"
+            field="field"
+            disabled="isDisabled"></macro-widget-toggle>
```
Comment (tonybach) [inline cdap-ui/app/hydrator/templates/partial/node-config-modal/node-config/plugin-edit-properties-template.html:56]: here too

File: cdap-ui/app/directives/widget-container/widget-wrangler-directives/widget-wrangler-directives.html [hunk 2]
```diff
@@ -14,6 +14,15 @@
...
 -->
 <div class="clearfix">
+  <textarea
+    class="form-control"
+    ng-trim="false"
+    ng-model="model"
+    rows="{{rows}}"
+    placeholder="{{placeholder}}"
+  >
+  </textarea>
```
Comment (tonybach) [inline cdap-ui/app/directives/widget-container/widget-wrangler-directives/widget-wrangler-directives.html:24]: nit: can just use a self-closing tag here

**Comment by tonybach** [comment]
Talked with @elfenheart , seems like if I set the schema to be a macro, it’s not being saved when i reopen it

macro-schema

File: cdap-ui/app/hydrator/templates/partial/node-config-modal/node-config/plugin-edit-properties-template.html (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/app/hydrator/templates/partial/node-config-modal/node-config/plugin-edit-properties-template.html:56]: this is not react, cannot use self closing tag 😂

Comment (tonybach) [reply cdap-ui/app/hydrator/templates/partial/node-config-modal/node-config/plugin-edit-properties-template.html:56]: oh forgot lol :joy:

**Comment by ajainarayanan** [comment]
Talked to @elfenheart . Noticed two things,

1. `ErrorDataset` in JS transform seems to be out of plugin properties and is rendered UI only. That is missing the tooltip behavior
2. Clicking on plugin properties on published pipeline view allows it to edit. May be (`macroWidgetToggle` directive should use `disabled` prop I guess)

**Comment by ajainarayanan** [comment]
Code 👍 LGTM

**Review by bdmogal — APPROVED** [review]

**Pull request merged**